# ==============================================================================
# MIT License
#
# Copyright (c) 2023-2024 Institute for Automotive Engineering (ika), RWTH Aachen University
# Copyright (c) 2024 Instituto de Telecomunicações, Universidade de Aveiro
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
# ==============================================================================

# --- Auto-generated by asn1ToRosMsg.py ----------------------------------------

# --- ASN.1 Definition ---------------------------------------------------------
# BasicVehicleContainerHighFrequency ::= SEQUENCE {
# 	heading Heading,
# 	speed Speed,
# 	driveDirection DriveDirection,
# 	vehicleLength VehicleLength,
# 	vehicleWidth VehicleWidth,
# 	longitudinalAcceleration LongitudinalAcceleration,
# 	curvature Curvature,
# 	curvatureCalculationMode CurvatureCalculationMode,
# 	yawRate YawRate,
# 	accelerationControl AccelerationControl OPTIONAL,
# 	lanePosition LanePosition OPTIONAL,
# 	steeringWheelAngle SteeringWheelAngle OPTIONAL,
# 	lateralAcceleration LateralAcceleration OPTIONAL,
# 	verticalAcceleration VerticalAcceleration OPTIONAL,
# 	performanceClass PerformanceClass OPTIONAL,
# 	cenDsrcTollingZone CenDsrcTollingZone OPTIONAL
# }
# ------------------------------------------------------------------------------
Heading heading

Speed speed

DriveDirection drive_direction

VehicleLength vehicle_length

VehicleWidth vehicle_width

LongitudinalAcceleration longitudinal_acceleration

Curvature curvature

CurvatureCalculationMode curvature_calculation_mode

YawRate yaw_rate

AccelerationControl acceleration_control
bool acceleration_control_is_present

LanePosition lane_position
bool lane_position_is_present

SteeringWheelAngle steering_wheel_angle
bool steering_wheel_angle_is_present

LateralAcceleration lateral_acceleration
bool lateral_acceleration_is_present

VerticalAcceleration vertical_acceleration
bool vertical_acceleration_is_present

PerformanceClass performance_class
bool performance_class_is_present

CenDsrcTollingZone cen_dsrc_tolling_zone
bool cen_dsrc_tolling_zone_is_present

